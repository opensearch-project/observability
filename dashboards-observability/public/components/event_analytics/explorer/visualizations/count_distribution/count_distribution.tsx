/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { BarOrientation, LONG_CHART_COLOR } from '../../../../../../common/constants/shared';
import { Plt } from '../../../../visualizations/plotly/plot';

export const CountDistribution = ({ countDistribution }: any) => {
  if (
    !countDistribution ||
    !countDistribution.data ||
    !countDistribution.metadata ||
    !countDistribution.metadata.fields
  )
    return null;

  const {
    data,
    metadata: { fields },
  } = countDistribution;

  const finalData = [
    {
      x: [...data[fields[1].name]],
      y: [...data[fields[0].name]],
      type: 'bar',
      name: fields[0],
      orientation: BarOrientation.vertical,
    },
  ];

  return (
    <Plt
      data={finalData}
      layout={{
        showlegend: true,
        margin: {
          l: 60,
          r: 10,
          b: 15,
          t: 30,
          pad: 0,
        },
        height: 220,
        colorway: [LONG_CHART_COLOR],
      }}
    />
  );
};
