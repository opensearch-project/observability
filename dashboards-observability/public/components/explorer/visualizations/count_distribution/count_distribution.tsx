/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { LONG_CHART_COLOR } from '../../../../../common/constants/shared';
import React from 'react';
import { Bar } from '../../../visualizations/charts/bar';

export const CountDistribution = ({ countDistribution }: any) => {
  if (
    !countDistribution ||
    !countDistribution.data ||
    !countDistribution.metadata ||
    !countDistribution.metadata.fields
  )
    return null;

  const layout = {
    showlegend: true,
    margin: {
      l: 60,
      r: 10,
      b: 15,
      t: 30,
      pad: 0,
    },
    height: 220,
    colorway: [LONG_CHART_COLOR],
  };

  return (
    <Bar
      visualizations={countDistribution}
      name="Event counts"
      layoutConfig={layout}
      isUniColor={true}
    />
  );
};
