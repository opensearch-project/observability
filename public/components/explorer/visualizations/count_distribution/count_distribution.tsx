/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { Bar } from '../../../visualizations/charts/bar';

export const CountDistribution = ({
  countDistribution
}: any) => {

  if (!countDistribution || !countDistribution.data) return null;
  
  const meta = countDistribution.metadata;
  const xkey = meta?.xfield?.name;
  const ykey = meta?.yfield?.name;
  const layout = {
    showlegend: true,
    margin: {
      l: 60,
      r: 10,
      b: 15,
      t: 30,
      pad: 0,
    },
    height: 220
  };
  const config = {};
  const xaxis = {
    autorange: true
  };
  const yaxis = {
    fixedrange: true
  };

  if (!xkey || !ykey) {
    return null;
  }

  return (
    <Bar
      visualizations={ countDistribution }
      name="Event counts"
      layoutConfig={ layout }
      config={ config }
    />
  );
};