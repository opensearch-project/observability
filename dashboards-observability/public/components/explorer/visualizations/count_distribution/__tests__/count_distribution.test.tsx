/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { CountDistribution } from '../count_distribution';
import { SAMPLE_VISUALIZATIONS } from '../../../../../../test/event_analytics_constants';

describe('Count distribution component', () => {
  configure({ adapter: new Adapter() });

  it('Renders empty count distribution component', async () => {
    
    const wrapper = mount(
      <CountDistribution />
    );
    
    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('Renders count distribution component with data', async () => {

    const wrapper = mount(
      <CountDistribution
        countDistribution={SAMPLE_VISUALIZATIONS}
      />
    );
    
    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});