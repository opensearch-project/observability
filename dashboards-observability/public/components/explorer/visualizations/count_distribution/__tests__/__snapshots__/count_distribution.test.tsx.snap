// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No result component Renders count distribution component with data 1`] = `
<CountDistribution
  countDistribution={
    Object {
      "count()": Array [
        2549,
        9337,
        1173,
      ],
      "jsonData": Array [
        Object {
          "count()": 2549,
          "span(timestamp,1M)": "2021-05-01 00:00:00",
        },
        Object {
          "count()": 9337,
          "span(timestamp,1M)": "2021-06-01 00:00:00",
        },
        Object {
          "count()": 2549,
          "span(timestamp,1M)": "2021-07-01 00:00:00",
        },
      ],
      "metadata": Object {
        "fields": Array [
          Object {
            "name": "count()",
            "type": "integer",
          },
          Object {
            "name": "span(timestamp,1M)",
            "type": "timestamp",
          },
        ],
      },
      "span(timestamp,1M)": Array [
        "2021-05-01 00:00:00",
        "2021-06-01 00:00:00",
        "2021-07-01 00:00:00",
      ],
    }
  }
/>
`;

exports[`No result component Renders empty count distribution component 1`] = `<CountDistribution />`;
